aaelf64/aaelf64.rst

```diff
@@ -1700,10 +1700,12 @@ sections are compatible with Branch Target Identification mechanism. An
 executable or shared object with this bit set is required to generate
 `Custom PLTs`_ with BTI instruction.
 
-``GNU_PROPERTY_AARCH64_FEATURE_1_PAC`` This indicates that all executable
-sections have Return Address Signing enabled. An executable or shared object
-with this bit set can generate `Custom PLTs`_ with a PAC
-instruction.
+``GNU_PROPERTY_AARCH64_FEATURE_1_PAC`` This indicates that all
+executable sections have been protected with Return Address Signing.
+Its use is optional, meaning that an ELF file where this flag is unset
+can still have Return Address signing enabled in some of or all its
+executable sections. If used, this flag simplifies the task of
```

Comment:
For the final sentence starting "If used". We normally don't add justification to the ABI documents, the argument normally goes that if you do it somewhere then you've got to do it everywhere.

I suggest having a think about whether it is necessary for the sentence to exist in terms of a reference specification? If it isn't then I suggest removing it. To be clear, I've not personally got a strong opinion that it must be removed, if at least one person wants to keep it, I'm happy to keep it.

Reply:
Thanks, I don't have a strong opinion either, happy to go either way. @DanielKristofKiss what do you think?

Diverging a bit from the topic, the Arm ARM does have a mechanism to distinguish between what is a rule and what is an explanation. I find it useful to understand the intent of the spec without restricting implementations. Might be worth considering something similar for the ABI (FYI @stuij). It would also make it easier for external partners to contribute to the spec. I don't think this imposes a requirement on us to add explanations everywhere, though. Maybe something to discuss offline.

Reply:
> We normally don't add justification to the ABI documents, the argument normally goes that if you do it somewhere then you've got to do it everywhere.
Normally we don't change the ABI in this way. I think it is important to describe the purpose of the flag. ( otherwise it could be just deprecated and dropped in this case).

In the ACLE spec the design docs hold additional information/motives/rationals besides the spec. (https://github.com/ARM-software/acle/tree/next-release/main/design_documents)

Reply:
There's certainly a case for supporting documents to contain additional information. In this particular case I don't want to hold up the change as it is only a sentence.